doc/tor.md

```diff
@@ -90,41 +105,44 @@ and open port 8333 on your firewall (or use -upnp).
-the user running bitcoind to the same group and setting permissions appropriately. On
-Debian-based systems the user running bitcoind can be added to the debian-tor group,
-which has the appropriate permissions. Before starting bitcoind you will need to re-login
-to allow debian-tor group to be applied. Otherwise you will see the following notice: "tor:
-Authentication cookie /run/tor/control.authcookie could not be opened (check permissions)"
-on debug.log.
-
-An alternative authentication method is the use
-of the `-torpassword=password` option. The `password` is the clear text form that
-was used when generating the hashed password for the `HashedControlPassword` option
-in the tor configuration file. The hashed password can be obtained with the command
-`tor --hash-password password` (read the tor manual for more details).
+Since Tor version 0.2.7.1, Bitcoin Core makes use of Tor's control socket API to
+create and destroy ephemeral onion services programmatically.
+
+This means that if Tor is running and proper authentication has been configured,
+Bitcoin Core automatically creates an onion service to listen on. The goal is to
+increase the number of available onion nodes.
+
+This feature is enabled by default if Bitcoin Core is listening (`-listen`) and
+it requires a Tor connection to work. It can be explicitly disabled with
+`-listenonion=0`. If it is not disabled, it can configured using the
+`-torcontrol` and `-torpassword` settings. To show verbose debugging
+information, pass `-debug=tor`.
+
+Connecting to Tor's control socket API requires one of two authentication
+methods to be configured. It also requires the control socket to be enabled,
+e.g. put `ControlPort 9051` in the `torrc` config file. For cookie
+authentication, the user running bitcoind must have read access to the
+`CookieAuthFile` specified in the Tor configuration. In some cases this is
+preconfigured and the creation of an onion service is automatic. If permission
+problems are seen in the debug log (use `-debug=tor` for this logging), they can
+be resolved by adding both the user running Tor and the user running bitcoind to
+the same group and setting permissions appropriately. On Debian-based systems,
+the user running bitcoind can be added to the debian-tor group, which has the
+appropriate permissions. Before starting bitcoind, you will need to log in again
+to allow the debian-tor group to be applied. Otherwise, you will see the
+following notice in the debug log: "tor: Authentication cookie
+/run/tor/control.authcookie could not be opened (check permissions)."
+
+An alternative authentication method is to use the `-torpassword=password`
+option. The `password` is the clear text form that was used when generating the
+hashed password for the `HashedControlPassword` option in the Tor configuration
+file. The hashed password can be obtained with the command `tor --hash-password
+password` (refer to the [Tor User Manual](https://tb-manual.torproject.org) for
```

Comment:
Hmm, https://tb-manual.torproject.org is the Tor _Browser_ manual. Would https://www.torproject.org/docs/tor-manual.html.en be more appropriate?

Reply:
> Hmm, 

or
https://2019.www.torproject.org/docs/tor-onion-service.html.en

Reply:
thanks, done